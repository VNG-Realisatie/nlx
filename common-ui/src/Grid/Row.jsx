import React, { PureComponent } from 'react'
import { Flex } from '@rebass/grid'

class Row extends PureComponent {
    render() {
        const { children } = this.props

        return <Flex {...this.props}>{children}</Flex>
    }
}

Row.propTypes = {}

Row.defaultProps = {
    mx: '-10px',
    flexWrap: 'wrap',
}

export default Row
