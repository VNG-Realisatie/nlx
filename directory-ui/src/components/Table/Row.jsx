// Copyright Â© VNG Realisatie 2018
// Licensed under the EUPL

import styled from 'styled-components'

const Row = styled.tr`
    display: table-row;

    &:first-child {
      td {
        &:first-child {
          border-top-left-radius: 3px;
        }

        &:last-child {
          border-top-right-radius: 3px;
        }
      }
    }

    &:not(:last-child) {
      td {
        border-bottom: 1px solid #F0F2F7;
      }
    }

    &:last-child {
      td {
        &:first-child {
          border-bottom-left-radius: 3px;
        }

        &:last-child {
          border-bottom-right-radius: 3px;
        }
      }
    }
`

export default Row
