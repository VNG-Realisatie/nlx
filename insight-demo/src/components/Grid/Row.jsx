import React, { PureComponent } from 'react'
import { Flex } from '@rebass/grid'

class Row extends PureComponent {
    render() {
        const { children } = this.props

        return <Flex {...this.props}>{children}</Flex>
    }
}

Row.propTypes = {}

Row.defaultProps = {
    mx: '-16px',
    flexWrap: 'wrap',
}

export default Row
