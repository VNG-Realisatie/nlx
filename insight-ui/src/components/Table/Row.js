import styled from 'styled-components'

const Row = styled.tr`
  display: table-row;

  &:first-child {
    td {
      &:first-child {
        border-top-left-radius: 3px;
      }

      &:last-child {
        border-top-right-radius: 3px;
      }
    }
  }

  &:not(:last-child) {
    td {
      border-bottom: 1px solid #f0f2f7;
    }
  }

  &:last-child {
    td {
      &:first-child {
        border-bottom-left-radius: 3px;
      }

      &:last-child {
        border-bottom-right-radius: 3px;
      }
    }
  }
`

export default Row
