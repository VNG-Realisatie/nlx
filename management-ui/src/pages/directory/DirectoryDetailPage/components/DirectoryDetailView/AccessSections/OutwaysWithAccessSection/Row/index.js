// Copyright Â© VNG Realisatie 2022
// Licensed under the EUPL
//
import React, { useContext, useState } from 'react'
import { arrayOf, instanceOf, string } from 'prop-types'
import { observer } from 'mobx-react'
import { useTranslation } from 'react-i18next'
import { Button, ToasterContext } from '@commonground/design-system'
import Table from '../../../../../../../../components/Table'
import { OutwayName, Outways } from '../../components/index.styles'
import AccessState from '../../components/AccessState'
import OutwayModel from '../../../../../../../../stores/models/OutwayModel'
import DirectoryServiceModel from '../../../../../../../../stores/models/DirectoryServiceModel'
import { IconClose } from '../../../../../../../../icons'
import { useConfirmationModal } from '../../../../../../../../components/ConfirmationModal'
import AccessDetails from './components/AccessDetails'
import { TdActions } from './index.styles'

const Row = ({ publicKeyFingerprint, outways, service }) => {
  const { t } = useTranslation()
  const { showToast } = useContext(ToasterContext)
  const [isLoading, setIsLoading] = useState(false)

  const { accessRequest, accessProof } =
    service.getAccessStateFor(publicKeyFingerprint)

  const [ConfirmTerminateModal, confirmTerminate] = useConfirmationModal({
    title: t('Terminate access'),
    okText: t('Terminate'),
    children: (
      <AccessDetails
        subTitle={t(
          'Terminating access will revoke access for the organization to the service. Are you sure?',
        )}
        organization={accessRequest.organization}
        serviceName={accessRequest.serviceName}
        publicKeyFingerprint={accessRequest.publicKeyFingerprint}
      />
    ),
  })

  const handleTerminate = async (evt) => {
    evt.stopPropagation()

    if (await confirmTerminate()) {
      try {
        setIsLoading(true)
        await service.terminateAccessRequest(accessRequest.publicKeyFingerprint)

        showToast({
          title: t('Access terminated'),
          variant: 'success',
        })
      } catch (err) {
        let message = err.message

        if (err.response && err.response.status === 403) {
          message = t(`You don't have the required permission.`)
        }

        showToast({
          title: t('Failed to terminate access'),
          body: message,
          variant: 'error',
        })
      }
    }
  }

  return (
    <Table.Tr>
      <Table.Td>
        <Outways>
          {outways.map((outway) => (
            <OutwayName key={outway.name}>{outway.name}</OutwayName>
          ))}
        </Outways>

        <AccessState
          isLoading={isLoading}
          accessRequest={accessRequest}
          accessProof={accessProof}
        />
      </Table.Td>

      <TdActions>
        <Button
          size="small"
          variant="link"
          onClick={handleTerminate}
          title={t(
            'Terminate access for Outways with public key fingerprint {{publicKeyFingerprint}}',
            {
              publicKeyFingerprint: accessRequest.publicKeyFingerprint,
            },
          )}
        >
          <IconClose inline />
          {t('Terminate')}
        </Button>

        <ConfirmTerminateModal />
      </TdActions>
    </Table.Tr>
  )
}

Row.propTypes = {
  publicKeyFingerprint: string,
  outways: arrayOf(instanceOf(OutwayModel)),
  service: instanceOf(DirectoryServiceModel),
}

export default observer(Row)
