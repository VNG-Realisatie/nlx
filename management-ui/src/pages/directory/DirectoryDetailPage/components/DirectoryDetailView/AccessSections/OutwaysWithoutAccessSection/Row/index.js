// Copyright Â© VNG Realisatie 2022
// Licensed under the EUPL
//
import React from 'react'
import { arrayOf, func, instanceOf, string } from 'prop-types'
import { useTranslation } from 'react-i18next'
import { observer } from 'mobx-react'
import Table from '../../../../../../../../components/Table'
import { OutwayName, Outways } from '../../components/index.styles'
import AccessState from '../../components/AccessState'
import OutwayModel from '../../../../../../../../stores/models/OutwayModel'
import DirectoryServiceModel from '../../../../../../../../stores/models/DirectoryServiceModel'
import { useConfirmationModal } from '../../../../../../../../components/ConfirmationModal'
import RequestAccessDetails from '../../components/RequestAccessDetails'

const Row = ({
  publicKeyFingerprint,
  outways,
  service,
  onShowConfirmRequestAccessModalHandler,
  onHideConfirmRequestAccessModalHandler,
}) => {
  const { t } = useTranslation()
  const [RequestConfirmationModal, confirmRequest] = useConfirmationModal({
    title: t('Request access'),
    okText: t('Send'),
    children: (
      <RequestAccessDetails
        service={service}
        publicKeyFingerprint={publicKeyFingerprint}
        outwayNames={outways.map((outway) => outway.name)}
      />
    ),
  })

  const onRequestAccess = async () => {
    onShowConfirmRequestAccessModalHandler()

    if (await confirmRequest()) {
      await service.requestAccess(publicKeyFingerprint)
    }

    onHideConfirmRequestAccessModalHandler()
  }

  const onRetryRequestAccess = () => {
    service.retryRequestAccess(publicKeyFingerprint)
  }

  const { accessRequest, accessProof } =
    service.getAccessStateFor(publicKeyFingerprint)

  return (
    <Table.Tr key={publicKeyFingerprint}>
      <Table.Td>
        <Outways>
          {outways.map((outway) => (
            <OutwayName key={outway.name}>{outway.name}</OutwayName>
          ))}
        </Outways>

        <AccessState
          accessRequest={accessRequest}
          accessProof={accessProof}
          onRequestAccess={onRequestAccess}
          onRetryRequestAccess={onRetryRequestAccess}
        />

        <RequestConfirmationModal />
      </Table.Td>
    </Table.Tr>
  )
}

Row.propTypes = {
  publicKeyFingerprint: string,
  outways: arrayOf(instanceOf(OutwayModel)),
  service: instanceOf(DirectoryServiceModel),
  onShowConfirmRequestAccessModalHandler: func,
  onHideConfirmRequestAccessModalHandler: func,
}

export default observer(Row)
