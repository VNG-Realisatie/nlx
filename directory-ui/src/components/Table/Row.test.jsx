import React from 'react'
import { shallow } from 'enzyme'
import Row from './Row'

it('should match snapshot', () => {
  const wrapper = shallow(<Row/>)
  expect(wrapper).toMatchSnapshot()
})

it('should render child elements', () => {
  const wrapper = shallow(<Row><th>Table head</th></Row>);
  expect(wrapper.contains(<th>Table head</th>)).toEqual(true)
})