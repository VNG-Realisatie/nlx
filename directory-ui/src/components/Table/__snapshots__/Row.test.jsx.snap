// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: table-row;
    
    &:first-child {
      td {
        border-top: 1px solid #EDEFF1;
      }
    
      td {
        &:first-child {
          border-top-left-radius: 3px;
          border-left: 1px solid #EDEFF1;
        }
        
        &:last-child {
          border-top-right-radius: 3px;
          border-right: 1px solid #EDEFF1;
        }
      }
    }
    
    &:not(:last-child) {
      td {
        border-bottom: 1px solid #EDEFF1;
      }
    }
    
    &:not(:first-child):not(:last-child) {
      td {
        &:first-child {
          border-left: 1px solid #EDEFF1;
        }
        
        &:last-child {
          border-right: 1px solid #EDEFF1;
        }
      }        
    }
    
    &:last-child {
      td {
        border-bottom: 1px solid #EDEFF1;
      
        &:first-child {
          border-bottom-left-radius: 3px;
          border-left: 1px solid #EDEFF1;
        }
        
        &:last-child {
          border-bottom-right-radius: 3px;
          border-right: 1px solid #EDEFF1;
        }
      }
    }
",
        ],
      },
      "displayName": "styled.tr",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "tr",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
